lib/utils/http.go

```diff
@@ -121,3 +121,20 @@ func GetAnyHeader(header http.Header, keys ...string) string {
 type HTTPDoClient interface {
 	Do(req *http.Request) (*http.Response, error)
 }
+
+// HTTPMiddleware defines a HTTP middleware.
+type HTTPMiddleware func(next http.Handler) http.Handler
+
+// ChainHTTPMiddlewares wraps an http.Handler with a list of middlewares. Inner
+// middlewares should be provided before outer middlewares.
+func ChainHTTPMiddlewares(next http.Handler, middlewares ...HTTPMiddleware) http.Handler {
```

Comment:
The `next` is the last object that is applied so maybe we can just rename it to `handler` or `topHandler` 
```suggestion
func ChainHTTPMiddlewares(hanlder http.Handler, middlewares ...HTTPMiddleware) http.Handler {
```